css-components/components-src/stylus/components/fab.styl

```diff
@@ -0,0 +1,187 @@
+
+// internal variables
+var-fab-width = 56px
+var-fab-height = 56px
+
+var-fab-mini-width = 40px
+var-fab-mini-height = 40px
+
+var-fab-background-color = $highlight-color
+var-fab-color = white
+var-fab-cta-background-color = $second-highlight-color
+var-border--fab-color = rgba(0, 0, 0, 0)
+var-fab-quiet-color = $highlight-color
+
+var-fab-cta-background-color = $second-highlight-color
+var-fab-cta-color = white
+var-fab-large-line-height = 36px
+var-fab-active-opacity = 0.2
+
+/*! topdoc
+  name: Fab
+  class: fab
+  modifiers:
+    :active: Active state
+    :disabled: Disabled state
+    :focus: Focused
+  markup:
+    <button class="fab">+</button>
+    <button class="fab" disabled>+</button>
+*/
+
+fab--disabled()
+  background-color alpha(black, 0.5)
+  shadow-0()
+  disabled()
+
+fab()
+  inline-block()
+  reset-box-model()
+  reset-base()
+  reset-font()
+  reset-cursor()
+  width var-fab-width
+  height var-fab-height
+  text-decoration none
+  font-size var-font-size
+  line-height var-fab-height
+  letter-spacing var-letter-spacing
+  color var-fab-color
+  vertical-align middle
+  text-align center
+  background-color alpha(var-fab-background-color, 1)
+  border 0px solid currentColor
+  border-radius 50%
+  shadow-2()
+  transition all 0.3s ease
+
+fab--focus()
+  outline 0
+
+fab--active()
+  shadow-3()
+  background-color alpha(var-fab-background-color, 1)
+  transition all 0.2s ease
+
+.fab
+  fab()
+
+.fab:hover
+  fab--hover()
+
+.fab:active
+  fab--active()
+
+.fab:focus
+  fab--focus()
+  
+.fab .fab-icon
+  overflow: hidden;
+  height: var-fab-height;
+  width: var-fab-width;
+
+.fab:disabled,
+.fab[disabled]
+  fab--disabled()
+  
+
+/*! topdoc
+  name: Fab with items
+  use: Fab
+  modifiers:
+    :active: Active state
+    :disabled: Disabled state
+    :focus: Focused
+  markup:
+    <div class="fab">+
+      <button class="fab fab-item">-</button>
+      <button class="fab fab-item">-</button>
+      <button class="fab fab-item">-</button>
+    </div>
+*/
+
+.fab .fab-item
+  width var-fab-mini-width
+  height var-fab-mini-height
+  shadow-2()
+  line-height var-fab-mini-height
+  position absolute
+  transform scale(0)
+  transition all 0.2s ease
+
+.fab
+  for num in 2 3 4 5 6 7
+    .fab-item:nth-child({num})
+      margin-top -(6px + 50px * num)
+      margin-left -21px
+      transition-delay: 25ms * (num - 1);
+      
+.fab
+  for num in 2 3 4 5 6 7
+    .fab-item--bottom:nth-child({num})
+      margin-top (10px + 50px * (num - 2))
+      margin-left -21px
+      transition-delay: 25ms * (num - 1);
+      
+.fab
+  for num in 2 3 4 5 6 7
+    .fab-item--left:nth-child({num})
+      margin-left -(30px + 50px * (num - 1))
+      margin-top -48px
+      transition-delay: 25ms * (num - 1);
+      
+.fab
+  for num in 2 3 4 5 6 7
+    .fab-item--right:nth-child({num})
+      margin-left (40px + 50px * (num - 2))
+      margin-top -48px
+      transition-delay: 25ms * (num - 1);
+
+.fab.fixed
```

Comment:
Request: Use modifier. Don't use multiple class. Follow the MindBEMding in css-components.